custom/icds_reports/utils/data_pull/Oct-2019.sql

```diff
@@ -0,0 +1,126 @@
+GROUP BY state_name
+
+
+/*
+# of AWCs that reported available medicine kit
+# of AWCs that reported unavailable medicine kit
+# of AWCs that reported available infant weighing scale
+# of AWCs that reported unavailable infant weighing scale
+# of AWCs that reported available mother and child weighing scale
+# of AWCs that reported unavailable mother and child weighing scale
+*/
+SELECT state_name,
+count(*) FILTER (WHERE infra_medicine_kits=1) AS "Available medicine kit",
+count(*) FILTER (WHERE infra_medicine_kits=0) AS "Unavailable medicine kit",
+count(*) FILTER (WHERE infra_baby_weighing_scale=1) AS "Available infant weighing scale",
+count(*) FILTER (WHERE infra_baby_weighing_scale=0) AS "Unavailable infant weighing scale",
+count(*) FILTER (WHERE infra_adult_weighing_scale=1) AS "Available mother and child weighing scale",
+count(*) FILTER (WHERE infra_adult_weighing_scale=0) AS "Unavailable mother and child weighing scale"
+FROM agg_awc_monthly
+WHERE aggregation_level=5 AND month='2019-10-01'
+GROUP BY state_name
+
+
+/*
+# of AWCs that reported available electricity line
+# of AWCs that reported unavailable electricity line
+# AWCs conducted at least 2 CBE events
+*/
+SELECT state_name,
+count(*) FILTER (WHERE electricity_awc=1) AS "Available electricity line",
+count(*) FILTER (WHERE electricity_awc=0) AS "Unavailable electricity line",
+sum(num_awcs_conducted_cbe) AS "AWCs conducted at least 2 CBE events",
+sum(num_awcs_conducted_vhnd) AS "AWCs conducted at least 1 VHNSD"
+FROM "public"."awc_location_months_local" "awc_location_months" LEFT JOIN "public"."agg_awc" "agg_awc" ON (
+        ("awc_location_months"."month" = "agg_awc"."month") AND
+        ("awc_location_months"."aggregation_level" = "agg_awc"."aggregation_level") AND
+        ("awc_location_months"."state_id" = "agg_awc"."state_id") AND
+        ("awc_location_months"."district_id" = "agg_awc"."district_id") AND
+        ("awc_location_months"."block_id" = "agg_awc"."block_id") AND
+        ("awc_location_months"."supervisor_id" = "agg_awc"."supervisor_id") AND
+        ("awc_location_months"."awc_id" = "agg_awc"."awc_id")
+    )
+WHERE "agg_awc"."aggregation_level"=5 AND "agg_awc"."month"='2019-10-01'
+GROUP BY state_name
+
+
+/*
+# Households Registered
+# Pregnant Women (should this use cases_ccs_pregnant or cases_ccs_pregnant_all)
+# Lactating Mothers (cases_ccs_lactating or cases_ccs_lactating_all)
+# Adolescent Girls (11-14y)
+*/
+SELECT state_name,
+SUM(cases_household) AS "Open Household Cases",
+SUM(cases_ccs_pregnant) AS "Pregnant",
+SUM(cases_ccs_lactating) AS "Lactating",
+Sum(cases_person_adolescent_girls_11_14) AS "Adolescent Girls (11-14y)"
+FROM agg_awc_monthly
+WHERE aggregation_level=1 AND month='2019-10-01'
+GROUP BY state_name
+
+
+-- # Children (0-6y)
+SELECT state_name,
+Count(*) FILTER (WHERE age_tranche::integer<=72 OR age_tranche IS null) AS "0-6y"
+FROM agg_child_health_monthly
```

Comment:
Hey this query will give you all the children age less than 6 year even if they are not valid in month.
basically these data pulls usually needs cases which are valid in the month. 
Specifically for this child count pull you can query agg_awc only and `cases_child_health` should give you that count.

Reply:
https://github.com/dimagi/commcare-hq/pull/25893/commits/6efa7a2317c2ffdd6aa2829cea268f09de90a453